File: server/CampaignTracker.Models/Entities/Campaign.cs
namespace CampaignTracker.Models;

public class Campaign : EntityBase 
{
    public string Name { get; set; }
    public string Description { get; set; }
    public string CampaignStart { get; set; } // string representations of dates
    public string CampaignEnd { get; set; } // string representations of dates
    public string Current { get; set; } //string representation of current date 
    public bool isComplete { get; set; }

    public IEnumerable<Session> Sessions { get; set ;}
    public IEnumerable<Character> Characters { get; set; }
}

File: client/core/models/api/campaign.ts
import { FormBuilder, FormGroup, Validators } from "@angular/forms";
import { EntityBase } from "../entity-base";
import { Character } from "./character";
import { Session } from "./session";

export interface Campaign extends EntityBase {
    name: string;
    description: string;
    campaignStart: string;
    campaignEnd: string;
    current: string;
    isComplete: boolean;

    sessions?: Session[];
    characters: Character[];
}

export const GenerateCampaignForm =  (campaign: Campaign, fb: FormBuilder): FormGroup => 
    fb.group({
        id: [campaign?.id],
        name: [campaign?.name, Validators.required],
        description: [campaign?.description],
        campaignStart: [campaign?.campaignStart],
        campaignEnd: [campaign?.campaignEnd],
        current: [campaign?.current],
        isComplete: [campaign?.isComplete]
    })